golem-rib/src/interpreter/rib_interpreter.rs

```diff
@@ -252,67 +250,71 @@ impl Interpreter {
                 }
                 RibIR::PushErrResult(analysed_type) => {
-                    internal::run_create_err_instruction(stack, analysed_type)?;
+                    internal::run_create_err_instruction(&mut stack, analysed_type)?;
                 }
                 RibIR::Concat(arg_size) => {
-                    internal::run_concat_instruction(stack, arg_size)?;
+                    internal::run_concat_instruction(&mut stack, arg_size)?;
                 }
                 RibIR::PushTuple(analysed_type, arg_size) => {
-                    internal::run_push_tuple_instruction(arg_size, analysed_type, stack)?;
+                    internal::run_push_tuple_instruction(arg_size, analysed_type, &mut stack)?;
                 }
                 RibIR::Negate => {
-                    internal::run_negate_instruction(stack)?;
+                    internal::run_negate_instruction(&mut stack)?;
                 }
 
                 RibIR::Label(_) => {}
 
                 RibIR::And => {
-                    internal::run_and_instruction(stack)?;
+                    internal::run_and_instruction(&mut stack)?;
                 }
 
                 RibIR::Or => {
-                    internal::run_or_instruction(stack)?;
+                    internal::run_or_instruction(&mut stack)?;
                 }
                 RibIR::ToIterator => {
-                    internal::run_to_iterator(stack)?;
+                    internal::run_to_iterator(&mut stack)?;
                 }
                 RibIR::CreateSink(analysed_type) => {
-                    internal::run_create_sink_instruction(stack, &analysed_type)?
+                    internal::run_create_sink_instruction(&mut stack, analysed_type)?
                 }
                 RibIR::AdvanceIterator => {
-                    internal::run_advance_iterator_instruction(stack)?;
+                    internal::run_advance_iterator_instruction(&mut stack)?;
                 }
                 RibIR::PushToSink => {
-                    internal::run_push_to_sink_instruction(stack)?;
+                    internal::run_push_to_sink_instruction(&mut stack)?;
                 }
 
                 RibIR::SinkToList => {
-                    internal::run_sink_to_list_instruction(stack)?;
+                    internal::run_sink_to_list_instruction(&mut stack)?;
                 }
 
                 RibIR::Length => {
-                    internal::run_length_instruction(stack)?;
+                    internal::run_length_instruction(&mut stack)?;
                 }
             }
         }
 
-        let stack_value = stack
-            .pop()
-            .unwrap_or_else(|| RibInterpreterStackValue::Unit);
-
-        let rib_result =
-            RibResult::from_rib_interpreter_stack_value(&stack_value).ok_or_else(no_result)?;
+        match byte_code_cursor.last() {
+            Some(RibIR::AssignVar(_)) => Ok(RibResult::Unit),
```

Comment:
An instruction that currently doesn't push anything to stack. This is tracked in a separate issue, probably this is fine for now.

https://github.com/golemcloud/golem/issues/1666

As reading, there are various ways to ensure stack containts atmost 1 value at the end, by using instructions like `Drop`, but not a concern of this PR